### PR TITLE
Add notes and questions for n19010

### PR DESCRIPTION
**Pull request by jnewbery**

### CODE REVIEW

File: _posts/2020-05-21-#19010.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  block in the requested range). The message handling code starts by deserializing
+  those fields into `uint_32t` and `uint256` local variables.
+
+- The `PrepareBlockFilterRequest()` helper function has its interface changed
+  to accept `start_height` and a `max_height_diff` parameters.
+
+- The `ProcessGetCFHeaders()` function calls a new method on the block filter
+  indexer class called `BlockFilterTypeName()`.
+
+- New tests are added to the `p2p_blockfilters.py` test file.
```
[the remark this answers is not in the document]
Comment (jonatack) [reply _posts/2020-05-21-#19010.md:36]: suggested continuation:

"as well as a class for each of the new messages in `test_framework/messages.py`, which are *(I'm guessing from the GH diff)* imported into `test_framework/mininode.py`. Note that all three of these files are in the `/test/functional/` directory in the repository root."

File: _posts/2020-05-21-#19010.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+layout: pr
+date: 2020-05-21
+title: "BIP 157: Add support for getcfheaders"
+pr: 19010
+authors: [jnewbery]
```
Comment (jonatack) [inline _posts/2020-05-21-#19010.md:6]: add jimpo as co-author in the authors array?

File: _posts/2020-05-21-#19010.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Notes
+
+- This PR adds support for the `getcfheaders` P2P message. You should familiarise
+  yourself with the format of that message and the `cfheaders` response before
+  reviewing the code changes. Both of those messages are defined in [BIP
+  157](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki#getcfilters).
+
+- The `getcfheaders` message contains two fields: `StartHeight` (The height of
+  the first block in the requested range) and `StopHash` (The hash of the last
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/2020-05-21-#19010.md:26]: These are defined as `StartHeight` and `StopHash` in the BIP (https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki#getcfheaders).

Good point about not capitalizing the definitions. I'll fix that.

File: _posts/2020-05-21-#19010.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  block in the requested range). The message handling code starts by deserializing
+  those fields into `uint_32t` and `uint256` local variables.
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/2020-05-21-#19010.md:28]: done

File: _posts/2020-05-21-#19010.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/2020-05-21-#19010.md:36]: I think it's safe to leave this as an exercise for the reader - any reviewer should be able to see that the last commit adds the tests, and work out where the new messages are defined and used in the test framework.

File: _posts/2020-05-21-#19010.md (hunk 2 above)
Comment (jnewbery) [reply _posts/2020-05-21-#19010.md:6]: Yes of course. Good catch.

**Comment by jnewbery** [comment]
Thanks for review, @jonatack. I've addressed or responded to all your comments.

I'll push this after the review club meeting tomorrow.

**Pull request merged**